src/main/java/carpet/mixins/RedstoneWireBlockMixin.java

```diff
@@ -64,96 +55,73 @@ private void onRedstoneWireBlockCTOR(Block.Settings settings, CallbackInfo ci) {
 
     /**
      * @author theosib, soykaf, gnembon
-     *
-    @Inject(method = "updateLogic", at = @At("HEAD"), cancellable = true)
-    private void updateLogicAlternative(World world, BlockPos pos, BlockState state, CallbackInfoReturnable<BlockState> cir)
-    {
-        if (CarpetSettings.fastRedstoneDust)
-            cir.setReturnValue(updateLogicPublic(world, pos, state));
+     */
+    @Inject(method = "update", at = @At("HEAD"), cancellable = true)
+    private void updateLogicAlternative(World world, BlockPos pos, BlockState state, CallbackInfo cir) {
+        if (CarpetSettings.fastRedstoneDust) {
+            updateLogicPublic(world, pos, state);
+            cir.cancel();
+        }
     }
 
     @Override
     public BlockState updateLogicPublic(World world_1, BlockPos blockPos_1, BlockState blockState_1) {
-        BlockState blockState_2 = blockState_1;
-        int int_1 = (Integer)blockState_1.get(POWER); // i
-        this.wiresGivePower = false;
-        int int_2 = world_1.getReceivedRedstonePower(blockPos_1); // k
-        this.wiresGivePower = true;
-        int int_3 = 0; // l
-        if (!CarpetSettings.fastRedstoneDust || int_2 < 15) {
-            Iterator var8 = Direction.Type.HORIZONTAL.iterator();
-
-            label43:
-            while(true) {
-                while(true) {
-                    if (!var8.hasNext()) {
-                        break label43;
-                    }
-
-                    Direction direction_1 = (Direction)var8.next();
-                    BlockPos blockPos_2 = blockPos_1.offset(direction_1);
-                    BlockState blockState_3 = world_1.getBlockState(blockPos_2);
-                    int_3 = this.increasePower(int_3, blockState_3);
-                    BlockPos blockPos_3 = blockPos_1.up();
-                    if (blockState_3.isSolidBlock(world_1, blockPos_2) && !world_1.getBlockState(blockPos_3).isSolidBlock(world_1, blockPos_3)) {
-                        int_3 = this.increasePower(int_3, world_1.getBlockState(blockPos_2.up()));
-                    } else if (!blockState_3.isSolidBlock(world_1, blockPos_2)) { //isSimpleFUllBLock
-                        int_3 = this.increasePower(int_3, world_1.getBlockState(blockPos_2.down()));
-                    }
-                }
-            }
-        }
-
-        int int_4 = int_3 - 1;
-        if (int_2 > int_4) int_4 = int_2;
-
-
-        if (int_1 != int_4) {
-            blockState_1 = (BlockState)blockState_1.with(POWER, int_4);
-            if (world_1.getBlockState(blockPos_1) == blockState_2) {
+        int i = this.getReceivedRedstonePower(world_1, blockPos_1);
+        BlockState blockState = blockState_1;
+        if (blockState_1.get(POWER) != i) {
+            blockState_1 = blockState_1.with(POWER, i);
+            if (world_1.getBlockState(blockPos_1) == blockState) {
                 world_1.setBlockState(blockPos_1, blockState_1, 2);
             }
 
             if (!CarpetSettings.fastRedstoneDust) {
```

Comment:
Ok - its easy to miss what's not highlighted. Yeah looks good to me.

I will let the challengers test it further but it looks to me like updated theo's redstone.

Reply:
Yes, the update method is only slightly modified, in this case to return the updated state. I found it to be the easiest way to get back to a successful implementation as without returning the updated state seems to be a bit more involved and would require modifying the redstone turbo class which I wanted to avoid as to not break anything.

Reply:
makes sense